.chglog/CHANGELOG.tpl.md

```diff
@@ -0,0 +1,22 @@
+{{ if .Versions -}}
+{{ range .Versions -}}
+{{- $chartName := regexReplaceAll "((?:[A-Za-z]*(?:\\-[A-Za-z]*)*))\\-(?:.*)" .Tag.Name "${1}" -}}
+{{- $chartversion := regexReplaceAll "(?:[A-Za-z]*(?:\\-[A-Za-z]*)*)\\-(.*)" .Tag.Name "v${1}" -}}
+# {{ $chartversion }}
+{{ range .CommitGroups -}}
+{{- if not (eq "Ignored" .Title ) -}}
+### {{ .Title }}
+{{ range .Commits -}}
+* {{ if .Scope }}**{{ .Scope }}** [{{ end }}[{{.Hash.Short}}]({{ $.Info.RepositoryURL }}/commit/{{ .Hash.Long }})]: {{ .Subject }}
```

Comment:
AFAICT from the regex below, `.Scope` will be the same for all charts, causing some weirdness in the changelogs.
![image](https://user-images.githubusercontent.com/6241450/205627236-be196035-dfc4-4685-8f2d-579d07c2f10c.png)

eg: commit `fix(node-analyzer,agent): something`, will `(node-analyzer,agent)`  be present in each `agent` and `node-analyzer` changelogs? 🤔 


We could fix this in two ways IMVHO:
1) Making the regex more complex, and grouping differently every substring in the `(...)`, comma separated, and match the chart name for the appropriate changelog message
2) Don't allow multi-chart changes 😄 Which makes probably more sense to me, given each chart will actually have slightly different changes (`sysdig-deploy` is usually just bumps in versions)


...or even totally ignore the weirdness and leave as it is - but I would rather choose one of the above.

If you have any other idea, please share @ironashram 🙏

Reply:
@ProvoK My initial idea was exactly your option 2, then @yashgiri  requested to support the multi chart `Scope`

The option 3 is that if the change affects multiple charts the `Scope` could be either avoided entirely or something generic in order to avoid weirdness in the changelogs

The thing that i would really like to avoid is over complicating the logic and the regex

Reply:
> The thing that i would really like to avoid is over complicating the logic and the regex

totally agree with this. Our life is hard enough even without regexes 🤣